src/hooks/useProductsList/handlers.js

```diff
@@ -0,0 +1,123 @@
+const initialState = {
+  list: null,
+  nextId: 0,
+  status: {
+    verb: '',
+    message: '',
+  },
+}
+
+export function loadList(items) {
+  items.sort((a, b) => (a.index > b.index ? 1 : -1))
+  const nextId = getSpareIdInList(items)
+  return {
+    nextId,
+    list: items,
+    status: {
+      verb: 'RESOLVED',
+      message: '',
+    },
+  }
+}
+
+export function addItem(state, newItem) {
+  const newList = [newItem, ...state.list]
+  const nextId = getSpareIdInList(newList)
+  return {
+    nextId,
+    list: newList,
+    status: {
+      verb: 'RESOLVED',
+      message: 'Produto adicionado com sucesso!',
+    },
+  }
+}
+
+export function addArray(state, array) {
+  const newList = [...array, ...state.list]
+  const nextId = getSpareIdInList(newList)
+  return {
+    ...state,
+    nextId,
+    list: newList,
+  }
+}
```

Comment:
Even though we will remove this `status` property from `state`  
Just to match other handlers dont spread state